### ISSUE TITLE
Feature Request: Pre/Post build phases

### ISSUE THREAD

**Issue by thii** [buildbuddy-io/rules_xcodeproj#1111]
This allows integrating linting/formatting tools into your Xcode project to get warnings and errors displayed in the issue navigator.

Unlike pre/post actions, additional build phases don't belong to a scheme, but to a target.

**Comment by brentleyjones** [comment]
> This allows integrating linting/formatting tools into your Xcode project to get warnings and errors displayed in the issue navigator.
>
> Unlike pre/post actions, additional build phases don't belong to schemes, but to targets.

I have some questions for you, @BalestraPatrick, and @erikkerber: Is this what these are going to be for? Just pre and post "build" (not target-compile) linting/formatting? I want to make sure I understand the full scope of what is asked for, since there seems to be some confusion in the discussion in #1117.

If linting, is there a reason it's not part of the build itself, using [Validation Actions](https://bazel.build/extending/rules#validation_actions)? If formatting, it will need to happen before `Bazel Build` so the changes are picked up before Bazel starts (so definitely not pre target-build). If it's something that needs to happen before or after a target builds that is tricky/nearly right now, and would require some BEP integration to get right (and it has to be non-source affecting per the previous statement).

**Comment by erikkerber** [comment]
Notably pre-build specifically for something like swiftlint, so the linting errors can be output into the Xcode IDE before any potentially long completion step. Aside from that, it's just a point in the build we can print output that Xcode renders inline as warnings/errors. This is not for any autofix system.

I'm assuming this use case won't work in a pre-scheme action, but I admit I've never tried either.

**Comment by brentleyjones** [comment]
>I'm assuming this use case won't work in a pre-scheme action, but I admit I've never tried either.

I believe Xcode still doesn't render the output from scheme pre-build actions, so this is correct.

**Comment by BalestraPatrick** [comment]
I believe the main use case is linting in the IDE for developers. We used to have this with Tulsi and lost the ability to run `swiftlint --reported xcode` from a build phase to show in-line violations. I was not aware of validation actions, and at the moment we have some scripts that run various linters manually.

**Comment by thii** [comment]
Let me break this down to get the feature we need in first:
* Pre-build with BwB (resolved by #1117)
* Post-build with BwB
* Pre-build with BwX
* Post-build with BwX

**Comment by thii** [comment]
> If linting, is there a reason it's not part of the build itself, using [Validation Actions](https://bazel.build/extending/rules#validation_actions)?

We still have the linting process running outside of the Bazel context, and only run it in the IDE.

**Comment by thii** [comment]
Re: Validations Actions
If anyone wants to get rid of your Xcode's pre-build phase, give [this](https://github.com/thii/rules_swiftlint) a try.
